indexing-hadoop/src/main/java/io/druid/indexer/IndexGeneratorJob.java

```diff
@@ -415,7 +431,7 @@ public int getPartition(BytesWritable bytesWritable, Writable value, int numPart
       final ByteBuffer bytes = ByteBuffer.wrap(bytesWritable.getBytes());
       bytes.position(4); // Skip length added by SortableBytes
       int shardNum = bytes.getInt();
-      if (config.get("mapred.job.tracker").equals("local")) {
+      if (local) {
```

Comment:
not sure if converting to a global instance variable is better, less global state is better.